engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1362,22 +1424,28 @@ private java.util.List<Tree> unrollApp(Tree list) {
     java.util.Collections.reverse(elems);
     return elems;
   }
-  private IR.Name qualifiedNameSegment(Tree tree, boolean generateId) {
+  private IR.Name qualifiedNameSegment(Tree tree, boolean generateId) throws SyntaxException {
     return switch (tree) {
       case Tree.Ident id -> sanitizeName(buildName(id, generateId));
       case Tree.Wildcard wild -> new IR$Name$Blank(getIdentifiedLocation(wild.getToken(), generateId), meta(), diag());
-      default -> throw new UnhandledEntity(tree, "qualifiedNameSegment");
+      default -> throw translateEntity(tree, "qualifiedNameSegment");
     };
   }
-  private List<IR.Name> qualifiedNameSegments(Tree t, boolean generateId) {
-    java.util.stream.Stream<IR.Name> segments =
-            unrollOprRhs(t, ".").stream().map(segment -> qualifiedNameSegment(segment, generateId));
-    return CollectionConverters.asScala(segments.iterator()).toList();
```

Comment:
Checked exceptions don't work particularly well with lambdas (the `@FunctionalInterface` would have to export the exception in signature, which is not common). As such I had to replace few `stream().map` with classical `for` cycles.